### PR TITLE
Tweaks to real PoH based on perf testing

### PR DESCRIPTION
**Pull request by pgarg66**
#### Problem
TPS is poor when real PoH is used.

#### Summary of Changes
1. Performance metrics for real PoH and lock contention
2. Reduced hash batch size 
3. TPS performance is slightly ~10% lower with current batch size. It gets worse with higher batch sizes.
4. Comment on CPU affinity change.

### CODE REVIEW

File: core/src/poh.rs [hunk 1]
```diff
@@ -32,6 +32,10 @@ impl Poh {
     }
 
     pub fn hash(&mut self, max_num_hashes: u64) -> bool {
+        if self.remaining_hashes == 1 {
```
Comment (mvines) [inline core/src/poh.rs]: Does this really help?

Comment (pgarg66) [reply core/src/poh.rs]: Not sure if it's helping. But it aligns well with ```record()```.

Comment (mvines) [reply core/src/poh.rs]: Hmm, but removing this new block doesn't change the behavior of `hash()` in any way so if it's not clearly helping...

**Comment by pgarg66** [comment]
https://github.com/solana-labs/solana/issues/4331

**Pull request merged**